src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -1779,6 +1803,7 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
             if (is_for_cpu == t_state.is_cpu_alloc) {
               transition(t_state, thread_state::THREAD_ALLOC_FREE);
             }
+            t_state.gpu_memory_allocated_bytes -= num_bytes;
```

Comment:
We also need to do the same thing here and only subtract the bytes allocated if it is for the GPU.